### PR TITLE
[ISSUE #12017] Add console backend configuration handling

### PR DESCRIPTION
**Pull request by RickonZhang0929**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

For #12017

Add console backend configuration handling with proxy and handler layers.

## Brief changelog

- Add ConfigHandler interface for defining configuration operations.
- Add ConfigInnerHandler to handle internal configuration logic.
- Add ConfigProxy to delegate configuration tasks based on deployment type.
- Add ConfigController to utilize ConfigProxy for configuration operations.
- Update ConsoleConfig to get deployment type.
- Update application.properties to add auth fields.

## Verifying this change

The changes can be verified by checking the configuration handling operations in the console backend, ensuring that the proxy and handler layers function correctly based on deployment type. Additionally, verify that the authentication fields in `application.properties` are correctly utilized.

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    }
+    
+    /**
+     * Get configure board information fail.
+     *
+     * @throws ServletException ServletException.
+     * @throws IOException      IOException.
+     * @throws NacosException   NacosException.
+     */
+    @GetMapping("/configs")
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:64]: configs可以抽到类的Mapping里

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Controller for handling HTTP requests related to configuration operations.
+ *
+ * @author zhangyukun
+ */
+@RestController
+@RequestMapping("/v3/console/cs/configs")
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:46]: uri需要用复数形式吗？

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        }
+    }
+    
+    /**
+     * Get the specific configuration information that the console USES.
+     *
+     * @throws NacosException NacosException.
+     */
+    @GetMapping("/detail")
+    public ResponseEntity<?> detailConfigInfo(@RequestParam("dataId") String dataId,
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:125]: API的返回内容最好使用规范化的RestResult或者com.alibaba.nacos.api.model.v2.Result

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 1 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: 已解决

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 2 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: v2使用单数形式，已修改

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 3 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: 我参考了原有代码修改返回类型

1.抛出Throwable
2.使用RestResultUtils记录成功和失败
3.使用LOGGER记录失败日志
4.根据名称填写失败信息msg

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    
+    /**
+     * Adds or updates non-aggregated data.
+     *
+     * @throws NacosException NacosException.
+     */
+    @PostMapping
+    public RestResult<Boolean> publishConfig(HttpServletRequest request, HttpServletResponse response,
+            @RequestParam("dataId") String dataId, @RequestParam("group") String group,
+            @RequestParam(value = "tenant", required = false, defaultValue = "") String tenant,
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:85]: 新接口最好统一用namespaceId替换tenant

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            @RequestParam(value = "schema", required = false) String schema,
+            @RequestParam(required = false) String encryptedDataKey) {
+        boolean result = false;
+        try {
+            result = configProxy.publishConfig(request, response, dataId, group, tenant, content, tag, appName, srcUser,
+                    configTags, desc, use, effect, type, schema, encryptedDataKey);
+            return RestResultUtils.success(result);
+        } catch (Throwable e) {
+            LOGGER.error("publish configuration information fail", e);
+            return RestResultUtils.failed(500, result, "publish configuration information fail");
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:103]: 错误码需要判断是否是已知的问题，比如NacosException或者NacosRuntimeException等。

如果这里不确定的话， 就不要做异常处理， 交给ExceptionHandler来统一处理。

**Review by KomachiSion — CHANGES_REQUESTED** [review]
Controller层一般情况下会要做的事：
1. 参数类型的转换和封装
2. 参数有效性的校验
3. 返回内容的转换和封装。

目前提交的Controller基本都没有做到。

可以参考一下前一个同学写的ConfigControllerV2。

因为你是新增了一套新的接口，后面UI控制台会按照你新定的API进行调用， 所以很多参数、规范不必拘泥于旧接口。可以按照标准的方式去设计。

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+            Pair<String, String> pair = EncryptionHandler.encryptHandler(dataId, content);
+            content = pair.getSecond();
+            encryptedDataKeyFinal = pair.getFirst();
+        }
+        
+        ParamUtils.checkTenant(namespaceId);
+        ParamUtils.checkParam(dataId, group, "datumId", content);
+        ParamUtils.checkParam(tag);
+        
+        ConfigForm configForm = new ConfigForm();
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:119]: 为什么不直接在方法上用ConfigForm？

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+            @RequestParam(required = false) String encryptedDataKey) throws NacosException {
+        // Validate parameters and convert the parameter type
+        String encryptedDataKeyFinal = null;
+        if (StringUtils.isNotBlank(encryptedDataKey)) {
+            encryptedDataKeyFinal = encryptedDataKey;
+        } else {
+            Pair<String, String> pair = EncryptionHandler.encryptHandler(dataId, content);
+            content = pair.getSecond();
+            encryptedDataKeyFinal = pair.getFirst();
+        }
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:113]: 解密应该算是业务逻辑了，其他的发布配置的时候也是在controller里面吗？

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+    
+    /**
+     * Get configure board information fail.
+     *
+     * @throws ServletException  ServletException.
+     * @throws IOException       IOException.
+     * @throws NacosApiException NacosApiException.
+     */
+    @GetMapping
+    @Secured(action = ActionTypes.READ, signType = SignType.CONFIG)
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:90]: 是不是少了AdminAPI的指定

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+    public void getConfig(HttpServletRequest request, HttpServletResponse response,
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java:91]: getConfig却不返回内容吗？ 是不是有错误？

File: console/src/main/resources/application.properties [hunk 10]
```diff
@@ -114,7 +114,9 @@ nacos.core.auth.system.type=nacos
 
 ### If turn on auth system:
 nacos.core.auth.enabled=false
-
+### If turn on auth system v3:
+nacos.auth.console.enabled=true
+nacos.auth.server.enabled=false
```
Comment (KomachiSion) [inline console/src/main/resources/application.properties:119]: 这个是鉴权的开关？

**Review by KomachiSion — CHANGES_REQUESTED** [review]
proxy和controller的命名看下是否调整一下， 有几个method和api我看着有点难理解。

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 4 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: 新的接口都已替换

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 5 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: controller这里就不再异常处理了

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 6 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: 已修改

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 7 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: 这部分下放到了handler，已修改

**Comment by RickonZhang0929** [comment]
> proxy和controller的命名看下是否调整一下， 有几个method和api我看着有点难理解。

命名已调整

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 8 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: 已加上

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java (hunk 9 above)
Comment (RickonZhang0929) [reply console/src/main/java/com/alibaba/nacos/console/controller/v3/ConsoleConfigController.java]: 已解决，这个接口更新了，替换了底层方法，有返回值了

File: console/src/main/resources/application.properties (hunk 10 above)
Comment (RickonZhang0929) [reply console/src/main/resources/application.properties]: 已删除，这是鉴权的开关，后面放到鉴权PR上提交

File: console/src/main/java/com/alibaba/nacos/console/controller/v3/config/ConsoleConfigController.java [hunk 11]
```diff
@@ -0,0 +1,409 @@
...
+     * @param configTags  Configuration tags.
+     * @param appName     Application name string value.
+     * @param pageNo      Page number.
+     * @param pageSize    Page size.
+     * @return Result containing the configuration information.
+     * @throws ServletException If a servlet-specific error occurs.
+     * @throws IOException      If an I/O error occurs.
+     * @throws NacosException   If a Nacos-specific error occurs.
+     */
+    @GetMapping
```
Comment (KomachiSion) [inline console/src/main/java/com/alibaba/nacos/console/controller/v3/config/ConsoleConfigController.java:94]: 一般来说， list接口用`$parentPath/list`, 下面的detail接口，及查询配置详情的接口，直接用GET 的`$parentPath` 比较好

**Review by KomachiSion — CHANGES_REQUESTED** [review]
最后一点小问题调整一下，

然后加一下单元测试。

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**